bodhi/server/util.py

```diff
@@ -730,21 +753,31 @@ def get_critpath_components_from_pdc(branch, component_type='rpm'):
     return list(critpath_pkgs_set)
 
 
-def api_get(api_url, service_name, error_key=None):
+def call_api(api_url, service_name, error_key=None, method='GET', **kwargs):
     """
-    A wrapper to get API resources with some error handling.
-    :param api_url: a string of the URL to query
+    A wrapper to send HTTP request with some error handling.
+    :param api_url: a string of the API URL.
     :param service_name: a string of the service name to use in error messages
     :param error_key: a string of the key that holds the error message in the
     JSON body. If this is set to None, the whole JSON will be used as the
     error message.
+    :param error_key: a string of the HTTP method to use to request. The defaut
+    value is ``GET``.
     :return: dictionary of the returned JSON or a RuntimeError
     """
-    base_error_msg = (
-        'Bodhi failed to get a resource from {0} at the following URL '
-        '"{1}". The status code was "{2}".')
-    rv = requests.get(api_url, timeout=60)
-
+    if method == 'POST':
+        base_error_msg = (
+            'Bodhi failed to send POST request to {0} at the following URL '
+            '"{1}". The status code was "{2}".')
+        rv = requests.post(api_url,
+                           headers={'Content-Type': 'application/json'},
+                           data=json.dumps(kwargs.get('data', {})),
```

Comment:
Since kwargs is only meaningful if it has a ```data``` in it, why not avoid the generic and undocumented ```kwargs``` variable, and instead give the method signature a ```post_params=None``` (or the name you are using ```data``` is fine too), which gets used here if set? Please do document this in the docblock.